addon/components/bs-form-element.js

```diff
@@ -505,5 +505,49 @@ export default FormGroup.extend({
       defineProperty(this, 'value', computed.alias(`model.${this.get('property')}`));
       this.setupValidations();
     }
-  }
+  },
+
+  /*
+   * adjust feedback icon position
+   *
+   * Bootstrap documentation:
+   *  Manual positioning of feedback icons is required for [...] input groups
+   *  with an add-on on the right. [...] For input groups, adjust the right
+   *  value to an appropriate pixel value depending on the width of your addon.
+   */
+  adjustFeedbackIcons: on('didInsertElement', observer('hasFeedback', 'formLayout', function() {
+    run.scheduleOnce('afterRender', () => {
+      // validation state icons are only shown if form element has feedback
+      if (this.get('hasFeedback') && !this.get('isDestroying')) {
+        // form group element has
+        this.$()
+          // an input-group
+          .has('.input-group')
+          // an addon or button on right side
+          .has('.input-group input + .input-group-addon, .input-group input + .input-group-btn')
+          // an icon showing validation state
+          .has('.form-control-feedback')
+          .each((i, formGroups) => {
+            // clear existing adjustment
+            this.$('.form-control-feedback').attr('style', '');
```

Comment:
CSP is a hard case. This line violates CSP if `style-src` does not allow `unsafe-inline`.